### PR TITLE
feat: add table tracking / remove keyboard shortcuts

### PR DESCRIPTION
**Pull request by cribbles**
Adds table tracking for insert and removal events, including `tablePaste`.

Does not include cypress tests, though I've confirmed this all works locally using the new `window.events` property in the demo environment.

Removes keyboard shortcuts (I'd add this in a separate PR but it would be too cumbersome to accomplish separately, given the code path overlap).

### CODE REVIEW

File: packages/rich-text/src/plugins/Table/index.tsx [hunk 1]
```diff
@@ -85,24 +86,55 @@ export const withTableOptions: CustomSlatePluginOptions = {
...
+function hasTables(nodes: CustomElement[]) {
+  const toCheck = ([] as CustomElement[]).concat(nodes);
+  while (toCheck.length) {
+    const { type, children } = toCheck.shift() as CustomElement;
+    if (type === BLOCKS.TABLE) {
+      return true;
+    } else if (children) {
+      // Ideally we'd simply spread here, but we'd need to
+      // upgrade TS first.
+      Array.prototype.push.apply(toCheck, children);
```
Comment (z0al) [inline packages/rich-text/src/plugins/Table/index.tsx:119]: What error do we get here?

File: packages/rich-text/src/RichTextEditor.tsx [hunk 2]
```diff
@@ -41,21 +41,22 @@ import {
   withEmbeddedEntityInlineOptions,
 } from './plugins/EmbeddedEntityInline';
 import { SdkProvider } from './SdkProvider';
+import { TrackingProvider, useTrackingContext } from './TrackingProvider';
 import { sanitizeIncomingSlateDoc, sanitizeSlateDoc } from './helpers/sanitizeSlateDoc';
 import { TextOrCustomElement } from './types';
 
 type ConnectedProps = {
   sdk: FieldExtensionSDK;
+  onAction: (name: string, data: Record<string, unknown>) => unknown;
```
Comment (z0al) [inline packages/rich-text/src/RichTextEditor.tsx:50]: Nit: I think having more specific typing for `name` would be ideal

**Review by z0al — APPROVED** [review]
LGTM. Left small comments/questions

File: packages/rich-text/src/plugins/Table/index.tsx (hunk 1 above)
Comment (cribbles) [reply packages/rich-text/src/plugins/Table/index.tsx:119]:
```
This syntax requires an imported helper named '__spreadArray' with 3
parameters, which is not compatible with the one in 'tslib'. Consider
upgrading your version of 'tslib'.ts
```

**Pull request merged**